packages/ledgerstate/signature.go

```diff
@@ -0,0 +1,343 @@
+package ledgerstate
+
+import (
+	"bytes"
+
+	"github.com/iotaledger/hive.go/byteutils"
+	"github.com/iotaledger/hive.go/crypto/bls"
+	"github.com/iotaledger/hive.go/crypto/ed25519"
+	"github.com/iotaledger/hive.go/marshalutil"
+	"github.com/iotaledger/hive.go/stringify"
+	"github.com/mr-tron/base58"
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/xerrors"
+)
+
+// region SignatureType ////////////////////////////////////////////////////////////////////////////////////////////////
+
+const (
+	// ED25519SignatureType represents an ED25519 Signature.
+	ED25519SignatureType SignatureType = iota
+
+	// BLSSignatureType represents a BLS Signature.
+	BLSSignatureType
+)
+
+// SignatureType represents the type of the signature scheme.
+type SignatureType uint8
+
+// String returns a human readable representation of the SignatureType.
+func (s SignatureType) String() string {
+	return [...]string{
+		"ED25519SignatureType",
+		"BLSSignatureType",
+	}[s]
+}
+
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region Signature ////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Signature is an interface for the different kind of Signatures that are support by the ledger state.
```

Comment:
```suggestion
// Signature is an interface for the different kinds of Signature that are supported by the ledger state.
```

Reply:
wouldnt it still be "different kinds of **Signatures**" (plural)?

Reply:
yep! 😛